content/security/rate-limiting.md

```diff
@@ -102,26 +102,30 @@ This module can work with websockets, but it requires some class extension. You
 ```typescript
 @Injectable()
 export class WsThrottlerGuard extends ThrottlerGuard {
-  async handleRequest(
-    context: ExecutionContext,
-    limit: number,
-    ttl: number,
-  ): Promise<boolean> {
+  async handleRequest(context: ExecutionContext, limit: number, ttl: number): Promise<boolean> {
     const client = context.switchToWs().getClient();
-    const ip = client.conn.remoteAddress;
+    // this is a generic method to switch between `ws` and `socket.io`. You can choose what is appropriate for you
+    const ip = ['conn', '_socket']
+      .map((key) => client[key])
+      .filter((obj) => obj)
+      .shift().remoteAddress;
```

Comment:
Rather than make this generic method, could we have the example be for `socket.io` and then in an `info` block underneath mention that for `ws` `_socket` should be used instead of `conn`? We don't do something like the above anywhere else in the docs (that I'm aware of) and it would lead to misunderstanding of what's happening here

Reply:
Sure